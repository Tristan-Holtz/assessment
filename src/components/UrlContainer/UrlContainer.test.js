import React from 'react';
import {
  UrlContainer,
  mapStateToProps,
  mapDispatchToProps
} from './UrlContainer';
import { shallow } from 'enzyme';
import { setUrls } from '../../actions';

describe('UrlContainer', () => {
  let wrapper;
  let mockProps;
  beforeEach(() => {
    mockProps = {
      setUrls: jest.fn(),
      urls: [
        {
          id: 1,
          long_url:
            'https://images.unsplash.com/photo-1531898418865-480b7090470f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=934&q=80',
          short_url: 'http://localhost:3001/useshorturl/1',
          title: 'Awesome photo'
        }
      ]
    };
    wrapper = shallow(<UrlContainer {...mockProps} />);
  });

  it('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object with urls', () => {
      const mockState = {
        urls: [
          {
            id: 1,
            long_url:
              'https://images.unsplash.com/photo-1531898418865-480b7090470f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=934&q=80',
            short_url: 'http://localhost:3001/useshorturl/1',
            title: 'Awesome photo'
          }
        ],
        nonsense: '',
        moreNonsense: true
      };

      const expected = {
        urls: [
          {
            id: 1,
            long_url:
              'https://images.unsplash.com/photo-1531898418865-480b7090470f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=934&q=80',
            short_url: 'http://localhost:3001/useshorturl/1',
            title: 'Awesome photo'
          }
        ]
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with setUrls', () => {
      const mockDispatch = jest.fn();
      const urls = [
        {
          id: 1,
          long_url:
            'https://images.unsplash.com/photo-1531898418865-480b7090470f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=934&q=80',
          short_url: 'http://localhost:3001/useshorturl/1',
          title: 'Awesome photo'
        }
      ];
      const actionToDispatch = setUrls(urls);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setUrls(urls);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
