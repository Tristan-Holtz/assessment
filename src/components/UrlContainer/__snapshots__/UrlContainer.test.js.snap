// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlContainer should render the component 1`] = `
<section>
  <div
    className="url"
  >
    <h3>
      Awesome photo
    </h3>
    <a
      href="http://localhost:3001/useshorturl/1"
      target="blank"
    >
      http://localhost:3001/useshorturl/1
    </a>
    <p>
      https://images.unsplash.com/photo-1531898418865-480b7090470f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=934&q=80
    </p>
  </div>
</section>
`;
